include/GafferScene/ShuffleAttributes.h

```diff
@@ -0,0 +1,78 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2020, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef GAFFERSCENE_SHUFFLEATTRIBUTES_H
+#define GAFFERSCENE_SHUFFLEATTRIBUTES_H
+
+#include "GafferScene/SceneElementProcessor.h"
+
+#include "Gaffer/ShufflePlug.h"
+
+namespace GafferScene
+{
+
+class GAFFERSCENE_API ShuffleAttributes : public SceneElementProcessor
```

Comment:
It's not something for you to deal with now, but I want to phase out SceneElementProcessor in favour of base classes dedicated to each particular type of processing (see comments in header). Unfortunately, we already have `AttributeProcessor` and it's nigh on useless because it wants to dictate what attributes get processed. We only have one subclass of AttributeProcessor in Gaffer (DeleteAttributes). Would you have any objection if I refactored (on master only, after merging) it so that AttributeProcessor became the base class for ShuffleAttributes, as well as ShaderAssignment, Attributes etc?

Reply:
Yeah, I'd seen `AttributeProcessor` and was annoyed I couldn't use it... that refactor would be nice